modules/cart/src/Form/AddToCartForm.php

```diff
@@ -81,6 +81,13 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       $form['#access'] = FALSE;
     }
 
+    // If our default variation is set, we alter the add to cart form.
+    if ($default_variation = \Drupal::request()->query->get('dv')) {
+      if (is_numeric($default_variation)) {
+        $form['purchased_entity']['widget'][0]['attributes']['field_variation']['#default_value'] = $default_variation;
```

Comment:
You'd want to move this to the new field widget, for product attributes. Should be able to check before it invokes its methods to discover the variation to be selected by default